internal/pkg/scheduler/api_test.go

```diff
@@ -0,0 +1,133 @@
+package scheduler_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/env"
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/scheduler"
+	"github.com/keboola/keboola-as-code/internal/pkg/testproject"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils"
+)
+
+func TestSchedulerApiCalls(t *testing.T) {
+	t.Parallel()
+	logger, _ := utils.NewDebugLogger()
+
+	project := testproject.GetTestProject(t, env.Empty())
+	project.SetState("empty.json")
+	storageApi := project.Api()
+	token := storageApi.Token().Token
+	hostName, _ := storageApi.GetSchedulerApiUrl()
+
+	// Get default branch
+	branch, err := storageApi.GetDefaultBranch()
+	assert.NoError(t, err)
+	assert.NotNil(t, branch)
+
+	configTarget := &model.ConfigWithRows{
+		Config: &model.Config{
+			ConfigKey: model.ConfigKey{
+				BranchId:    branch.Id,
+				ComponentId: "ex-generic-v2",
+			},
+			Name:              "Test",
+			Description:       "Test description",
+			ChangeDescription: "My test",
+			Content: utils.PairsToOrderedMap([]utils.Pair{
+				{
+					Key: "foo",
+					Value: utils.PairsToOrderedMap([]utils.Pair{
+						{Key: "bar", Value: "baz"},
+					}),
+				},
+			}),
+		},
+	}
+	resConfigTarget, err := storageApi.CreateConfig(configTarget)
+	assert.NoError(t, err)
+	assert.Same(t, configTarget, resConfigTarget)
+	assert.NotEmpty(t, configTarget.Id)
+
+	configScheduler := &model.ConfigWithRows{
+		Config: &model.Config{
+			ConfigKey: model.ConfigKey{
+				BranchId:    branch.Id,
+				ComponentId: "keboola.scheduler",
+			},
+			Name:              "Test",
+			Description:       "Test description",
+			ChangeDescription: "My test",
+			Content: utils.PairsToOrderedMap([]utils.Pair{
+				{
+					Key: "schedule",
+					Value: utils.PairsToOrderedMap([]utils.Pair{
+						{Key: "cronTab", Value: "*/2 * * * *"},
+						{Key: "timezone", Value: "UTC"},
+						{Key: "state", Value: "disabled"},
+					}),
+				},
+				{
+					Key: "target",
+					Value: utils.PairsToOrderedMap([]utils.Pair{
+						{Key: "componentId", Value: "ex-generic-v2"},
+						{Key: "configurationId", Value: configTarget.Id},
+						{Key: "mode", Value: "run"},
+					}),
+				},
+			}),
+		},
+	}
+	resConfigScheduler, err := storageApi.CreateConfig(configScheduler)
+	assert.NoError(t, err)
+
+	api := scheduler.NewSchedulerApi(hostName, token, context.Background(), logger, true)
+
+	// List
+	schedules, err := api.ListSchedules()
+	assert.NoError(t, err)
+	schedulesLength := len(schedules)
```

Comment:
pouzil by som `assert.Len`

Reply:
No jo, ale já právě nevím kolik jich tam už je předtím než test začne. Respektive začal jsem s `assert.Len(t. schedules, 0)` a překvapilo mě že už tam nějaký schedules jsou. 🙂 A nevím přesně jak to poskytování projektů testama funguje, tak jsem to napsal takto "bezpečněji". Nicméně potom se mi to večer rozleželo a řekl jsem si, že je to stejně takový hloupý tady kontrolovat jen tu délku na seznamu schedules, tak jsem si řekl že tam ještě dodělám `GET /schedules/{id}` a budu to testovat přímo na něm.

Reply:
Respektive, ti to tu mozem rovno napisat:
- v subore `*_test.go`, pouzijes namiesto `package abc`, .... `package abc_test`
- pridas tam import `. "/path/to/abc"`, ... vsimni su tu `.` namiesto aliasu, to znamena ze v balicku `abc_test` budu dostupne vsetky datove typy z `abc` bez toho, aby bolo potrebne pouzit `abc.` ... tj, ako keby boli z `abc_test`
- jedina nevyhoda je, ze takto nemas pristup k `private` struktura, ale to v tvojom teste tusim nepotrebujes, kedze testujes iba `public` metody

Priklad:
https://github.com/keboola/keboola-as-code/blob/550c191a9a066d2bd43da83e63e445e5a9bd3e5d/internal/pkg/mapper/transformation/load_test.go#L1

https://github.com/keboola/keboola-as-code/blob/550c191a9a066d2bd43da83e63e445e5a9bd3e5d/internal/pkg/mapper/transformation/load_test.go#L12

Idealne je, sa cyklickym zavislostiam v teste vyhnut, ... ale niekedy to nejde.

Reply:
Tak povedlo se: https://github.com/keboola/keboola-as-code/commit/2c732126fb79eadf58c742cd95d4b7a6638e2fdf